### PR TITLE
[full-ci] fix docker dev env and decrease build time

### PR DESCRIPTION
**Pull request by fschade**
## Description
We've introduced a broken build build pipeline which had led to long waiting times during development. 

This pr fixes:
* re-build time increased on every cycle
* docker on oc10 stopped working and got stuck on file-change

## Related Issue
---

## Motivation and Context
I want to use my time wisely ;)

## How Has This Been Tested?
- local dev env on macOs and latest docker

## Screenshots (if appropriate):
---

## Types of changes
- [X] Bug fix (non-breaking change which fixes an issue)
- [X] Technical debt

## Checklist:
---

## Open tasks:
- [ ] Test on different OS

### CODE REVIEW

File: changelog/unreleased/bugfix-build-time [hunk 1]
```diff
@@ -0,0 +1,6 @@
+Bugfix: Decrease build time and fix faulty oc10 docker dev env.
```
Comment (kulmann) [inline changelog/unreleased/bugfix-build-time:1]:
```suggestion
Bugfix: Decrease build time and fix faulty oc10 docker dev env
```

according to log output the `.` at the end of the title is not allowed 👀

**Review by kulmann — COMMENTED** [review]
Not sure what I need to do to not see this `Oops. Something went wrong.` page from hell anymore with this PR. 🤔 Anything to consider? Like still something to be done as first step? I tried with running `yarn build:w` first and after that running `docker compose up -d`. On first build it worked, killing the watcher and rebuilding results in the error page again. Docker env was cleaned before, up to date images pulled.

**Comment by fschade** [comment]
> images

nice edgecase, thx i haven't tried to kill the rollup watch while docker keeps running...
i only tested the default development flow (change files). 

Ill give it a try tomorrow :)

**Comment by ownclouders** [comment]
Results for <strong>oCISSharingPublic1</strong> https://drone.owncloud.com/owncloud/web/25131/68/1
The following scenarios passed on retry:
- webUISharingPublicBasic/publicLinkEdit.feature:57
:boom: The acceptance tests pipeline failed. The build has been cancelled.

**Comment by dschmidt** [comment]
> Not sure what I need to do to not see this `Oops. Something went wrong.` page from hell anymore with this PR. thinking Anything to consider? Like still something to be done as first step? I tried with running `yarn build:w` first and after that running `docker compose up -d`. On first build it worked, killing the watcher and rebuilding results in the error page again. Docker env was cleaned before, up to date images pulled.

Haven't tried with this PR, but I'm using `docker-compose up -d --force-recreate` that usually works reliably for me

**Comment by fschade** [comment]
@kulmann, can you give it another try, i found out that some caching is going on.
Please test wit `disabled cache` in console network tab.
Bildschirmfoto 2022-05-03 um 12 10 50

**Comment by kulmann** [comment]
@fschade and @dschmidt that's what I already have, starting the docker env with `--force-recreate` always. And `Disable cache` in the browser network tab is always ticked since forever. Still the same error page. Browser console prints

```
index.html:19          GET http://host.docker.internal:8080/index.php/apps/web/js/require.js?1651573570081 net::ERR_ABORTED 404 (Not Found)
```

**Comment by fschade** [comment]
@kulmann lets try it out together if time permits....

**Review by kulmann — APPROVED** [review]
works for me 🤩

**Comment by kulmann** [comment]
I cancelled the drone job https://drone.owncloud.com/owncloud/web/25231 since it passed the e2e tests. As this PR doesn't change any code it's sufficient for me to know that **any** end to end testing succeeded. Which it did. Force merging now.

**Pull request merged**